### PR TITLE
Fix test filtering when using single-instrumentation-call (#526)

### PR DESCRIPTION
**Pull request by JDFind**
The main purpose of these changes are to fix Issue #526.
I think it also adds the functionality requested in #509.

### CODE REVIEW

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java [hunk 1]
```diff
@@ -298,52 +306,18 @@ private LogRecordingTestRunListener queryTestSet(final String testPackage,
...
-    }
-    if (test != null) {
-      runner.setMethodName(test.getClassName(), test.getTestName());
-    }
-    runner.run(listeners);
-  }
-
+  /**
+   * Cratea a configured Test Runner
+   * This method adds sharding, class name, method name, test size and coverage, if available
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: These two sentences are missing periods.

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java [hunk 2]
```diff
@@ -298,52 +306,18 @@ private LogRecordingTestRunListener queryTestSet(final String testPackage,
...
+   * @throws Exception
```
Comment (JakeWharton) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: Delete

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java (hunk 1 above)
Comment (JDFind) [reply spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: done

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java (hunk 2 above)
Comment (JDFind) [reply spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: done

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java [hunk 3]
```diff
@@ -229,21 +228,29 @@ public DeviceResult run(AndroidDebugBridge adb) {
...
-      logDebug(debug, "Running all tests in a single instrumentation call [%s]", serial);
       try {
-        runAllTestOnDevice(testPackage, testRunner, device, listeners);
+        logDebug(debug, "Running all tests in a single instrumentation call [%s]", serial);
+
+        RemoteAndroidTestRunner runner = createConfiguredRunner(testPackage, testRunner, device);
+        runner.run(listeners);
       } catch (Exception e) {
-        result.addException(e);
+          result.addException(e);
```
Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: revert

Comment (JDFind) [reply spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: done

File: spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java [hunk 4]
```diff
@@ -298,52 +305,17 @@ private LogRecordingTestRunListener queryTestSet(final String testPackage,
...
-      addCodeCoverageInstrumentationArgs(runner, device);
-    }
-    if (test != null) {
-      runner.setMethodName(test.getClassName(), test.getTestName());
-    }
-    runner.run(listeners);
-  }
-
+  /**
+   * Crate a configured Test Runner.
```
Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: One more: `Crate`->`Create`

Comment (JDFind) [reply spoon-runner/src/main/java/com/squareup/spoon/SpoonDeviceRunner.java]: How did I messed that up?
Done

**Comment by edenman** [comment]
Thanks @JDFind !

**Pull request merged**